src/box/gc.c

```diff
@@ -735,6 +809,238 @@ gc_consumer_name(struct gc_consumer *consumer)
 			  tt_uuid_str(&consumer->uuid));
 }
 
+/**
+ * Write the consumer to persistent WAL GC state.
+ */
+static int
+gc_consumer_persist(struct gc_consumer *consumer)
+{
+	if (!gc_has_persistent_consumers())
+		return 0;
+
+	/* Mark consumer as synchronized with persistent state. */
+	consumer->is_synced = true;
+	char tuple_buf[UUID_STR_LEN + VCLOCK_STR_LEN_MAX + GC_NAME_MAX + 30];
+	char *tuple_end = tuple_buf;
+	tuple_end = mp_encode_array(tuple_end, 3);
+	tuple_end = mp_encode_str0(tuple_end, tt_uuid_str(&consumer->uuid));
+	tuple_end = mp_encode_vclock_ignore0(tuple_end, &consumer->vclock);
+	tuple_end = mp_encode_map(tuple_end, 1);
+	tuple_end = mp_encode_str0(tuple_end, "object");
+	tuple_end = mp_encode_str0(tuple_end,
+				   gc_consumer_type_strs[consumer->type]);
+	assert((unsigned long)(tuple_end - tuple_buf) < sizeof(tuple_buf));
+
+	return box_replace(BOX_GC_CONSUMERS_ID, tuple_buf, tuple_end, NULL);
+}
+
+int
+gc_consumer_sync(struct gc_consumer *consumer)
+{
+	/*
+	 * We cannot persist consumer asynchronously for the first time
+	 * because hanging consumer can appear in the case of fail. So let's
+	 * always do a direct replace here.
+	 */
+	struct txn *txn = txn_begin();
+	if (txn == NULL)
+		return -1;
+	/*
+	 * No need to wait limbo, so let's set TXN_FORCE_ASYNC.
+	 * Moreover, the function can be called when quorum cannot be collected,
+	 * (for example, on IPROTO_SUBSCRIBE), so we must set this flag here.
+	 */
+	txn_set_flags(txn, TXN_FORCE_ASYNC);
+	if (gc_consumer_persist(consumer) != 0) {
```

Comment:
Nit: Let's name this helper function `gc_consumer_sync_impl`.

Reply:
Done.